lib/tesla/builder.ex

```diff
@@ -2,7 +2,26 @@ defmodule Tesla.Builder do
   @http_verbs ~w(head get delete trace options post put patch)a
   @body ~w(post put patch)a
 
+  require Logger
+
   defmacro __using__(opts \\ []) do
+    unless Application.get_env(:tesla, :warning_acknowledged, false) do
+      Logger.warn("""
+      `use Tesla.Builder` and `use Tesla` are soft-deprecated. It will be removed in future major version in favor of
```

Comment:
How about [`IO.warn`](https://hexdocs.pm/elixir/IO.html#warn/1) to avoid the dependency? Also, I think warning + removal is a regular deprecation, not a soft one.

```suggestion
      IO.warn("""
      `use Tesla.Builder` and `use Tesla` are deprecated. It will be removed in future major version in favor of
```